src/home-components/ArticleCardContainer.jsx

```diff
@@ -1,19 +1,53 @@
 import { useEffect, useState } from "react";
 import { getArticles } from "../api";
 import { ArticleCard } from "./ArticleCard";
+import { Link } from "react-router-dom";
+import { ArticleContainer } from "../articleView-components/ArticleContainer";
 export function ArticleCardContainer({ articleTopic }) {
   const [articles, setArticles] = useState([]);
+  const [isLoading, SetIsLoading] = useState(false);
+  const [isError, SetIsError] = useState(false);
   useEffect(() => {
+    SetIsLoading(true);
+    SetIsError(false);
     getArticles()
       .then((articles) => {
         setArticles(articles);
-        console.log(articles);
+        SetIsLoading(false);
+        SetIsError(false);
       })
-      .catch((err) => console.log(err));
+      .catch((err) => {
+        SetIsLoading(false);
+        SetIsError(true);
+        console.log(err);
+      });
   }, [getArticles]);
+
+  if (isLoading) {
+    return (
+      <img
+        className="loader"
+        src="Spinner@1x-1.0s-200px-200px.gif"
```

Comment:
This image I think needs to be in the src folder not public, its giving me a warning in gitpod.